### PR TITLE
[PT-98]-NextConfigError

### PR DESCRIPTION
**Pull request by ItsFaizan**
# [PT-98](https://0xbase.atlassian.net/browse/PT-98?atlOrigin=eyJpIjoiNTJkZTU3YWQzZGFiNDkyZTlkZDdmYTNkMDg4MDRiMzAiLCJwIjoiaiJ9) - NextConfigError

### Standalone Error fixed 

[PT-98]: https://0xbase.atlassian.net/browse/PT-98?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/drawer/Drawer.tsx [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+          Consequat mauris nunc congue nisi vitae suscipit. Fringilla est ullamcorper eget nulla facilisi etiam
+          dignissim diam. Pulvinar elementum integer enim neque volutpat ac tincidunt. Ornare suspendisse sed nisi lacus
+          sed viverra tellus. Purus sit amet volutpat consequat mauris. Elementum eu facilisis sed odio morbi. Euismod
+          lacinia at quis risus sed vulputate odio. Morbi tincidunt ornare massa eget egestas purus viverra accumsan in.
+          In hendrerit gravida rutrum quisque non tellus orci ac. Pellentesque nec nam aliquam sem et tortor. Habitant
+          morbi tristique senectus et. Adipiscing elit duis tristique sollicitudin nibh sit. Ornare aenean euismod
+          elementum nisi quis eleifend. Commodo viverra maecenas accumsan lacus vel facilisis. Nulla posuere
+          sollicitudin aliquam ultrices sagittis orci a.
+        </Typography>
+      </Box>
```
Comment (fuddin95) [inline src/components/drawer/Drawer.tsx:213]: You can remove this and render {children } here

File: src/components/drawer/Drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+interface Props {
+  /**
+   * Injected by the documentation to work in an iframe.
+   * Remove this when copying and pasting into your project.
+   */
+  window?: () => Window
+}
+
+export default function ResponsiveDrawer(props: Props) {
```
Comment (fuddin95) [inline src/components/drawer/Drawer.tsx:74]: please export a react functional component using arrow syntax like before

File: src/components/drawer/Drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+  const { window } = props
```
Comment (fuddin95) [inline src/components/drawer/Drawer.tsx:75]: you can destructure window props in function like so

File: src/components/drawer/Drawer.tsx [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+    setIsClosing(false)
+  }
+
+  const handleDrawerToggle = () => {
+    if (!isClosing) {
+      setMobileOpen(!mobileOpen)
+    }
+  }
+
+  const drawer = (
```
Comment (fuddin95) [inline src/components/drawer/Drawer.tsx:94]: you can name this `drawerItems`

File: src/app/dashboard/page.tsx [hunk 5]
```diff
@@ -1,7 +1,9 @@
+import PermanentDrawerLeft from '@/components/drawer/Drawer'
```
Comment (fuddin95) [inline src/app/dashboard/page.tsx:1]: Your function name is ResponsiveDrawer not `PermanentDrawerLeft`

File: src/components/drawer/Drawer.tsx [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    },
+  },
+}))
+
+interface Props {
+  /**
+   * Injected by the documentation to work in an iframe.
+   * Remove this when copying and pasting into your project.
+   */
+  window?: () => Window
```
Comment (fuddin95) [inline src/components/drawer/Drawer.tsx:71]: window is a global variable no need in props

File: src/components/drawer/Drawer.tsx [hunk 7]
```diff
@@ -0,0 +1,216 @@
...
+      </List>
+    </div>
+  )
+
+  // Remove this const when copying and pasting into your project.
+  const container = window !== undefined ? () => window().document.body : undefined
+
+  return (
+    <Box sx={{ display: 'flex' }}>
+      <CssBaseline />
```
Comment (fuddin95) [inline src/components/drawer/Drawer.tsx:127]: why do we need `<CssBaseline />` here?

File: src/components/drawer/Drawer.tsx [hunk 8]
```diff
@@ -0,0 +1,216 @@
...
+            FOLIO NOMICS
+          </Typography>
+          <Search>
+            <SearchIconWrapper>
+              <SearchIcon />
+            </SearchIconWrapper>
+            <StyledInputBase placeholder="Search…" inputProps={{ 'aria-label': 'search' }} />
+          </Search>
+        </Toolbar>
+      </AppBar>
```
Comment (fuddin95) [inline src/components/drawer/Drawer.tsx:157]: Lets seperate the app bar in a seperate file in the same folder maybe `DrawerAppBar.tsx`

File: src/components/drawer/Drawer.tsx [hunk 9]
```diff
@@ -0,0 +1,216 @@
...
+            color="inherit"
+            aria-label="open drawer"
+            edge="start"
+            onClick={handleDrawerToggle}
+            sx={{ mr: 2, display: { sm: 'none' } }}
+          >
+            <MenuIcon />
+          </IconButton>
+          <Typography variant="h6" noWrap component="div" sx={{ flexGrow: 1 }}>
+            FOLIO NOMICS
```
Comment (fuddin95) [inline src/components/drawer/Drawer.tsx:148]: Title should be a prop

File: src/app/dashboard/page.tsx [hunk 10]
```diff
@@ -1,7 +1,9 @@
...
+
 export default function Dashboard() {
   return (
     <main>
-      <h1>Dashboard</h1>
+      <PermanentDrawerLeft />
```
Comment (fuddin95) [inline src/app/dashboard/page.tsx:6]: Render Lorem lapsum here as  children component

**Review by fuddin95 — CHANGES_REQUESTED** [review]

File: src/components/drawer/Drawer.tsx (hunk 1 above)
Comment (ItsFaizan) [reply src/components/drawer/Drawer.tsx:213]: Resolved

File: src/components/drawer/Drawer.tsx (hunk 2 above)
Comment (ItsFaizan) [reply src/components/drawer/Drawer.tsx:74]: Done

File: src/components/drawer/Drawer.tsx (hunk 4 above)
Comment (ItsFaizan) [reply src/components/drawer/Drawer.tsx:94]: Done

File: src/app/dashboard/page.tsx (hunk 5 above)
Comment (ItsFaizan) [reply src/app/dashboard/page.tsx:1]: Done

File: src/components/drawer/Drawer.tsx (hunk 7 above)
Comment (ItsFaizan) [reply src/components/drawer/Drawer.tsx:127]: Removed it came from Mui Webiste we already have it in layout.tsx

File: src/components/drawer/Drawer.tsx (hunk 8 above)
Comment (ItsFaizan) [reply src/components/drawer/Drawer.tsx:157]: Done

File: src/app/dashboard/page.tsx (hunk 10 above)
Comment (ItsFaizan) [reply src/app/dashboard/page.tsx:6]: Done

File: src/components/drawer/Drawer.tsx (hunk 3 above)
Comment (ItsFaizan) [reply src/components/drawer/Drawer.tsx]: Resolved

File: src/components/drawer/Drawer.tsx (hunk 6 above)
Comment (ItsFaizan) [reply src/components/drawer/Drawer.tsx]: Done

**Pull request merged**